### PR TITLE
WIP: IoT Hub -> AZ Func -> CrateDB Architecture Reference

### PR DESCRIPTION
**Pull request by autophagy**
@mechanomi Here's a draft WIP PR for the first architecture reference document, concerning IoT Hub -> CrateDB via Azure Functions.

Before we commit to properly editing it and making sure it fits within our stylistic constraints, i'd like to get your feedback on a few structural elements, namely:

- The naming of this document, and its location within the documentation structure.
- Whether you think this structure of having a high-level abstract overview of an architecture (with an architecture diagram of some sort (i just whipped up a quick one as an example)), followed by a concrete implementation of that architecture with code samples makes sense for what we envision for this section.

### CODE REVIEW

**Comment by jayeff** [comment]
FYC: @kamejosh published a Github Repo with an Azure Function to ingest EventHub messages into CrateDB. There is some overlap to this reference architecture documentation.
https://github.com/crate/eventhub-consumer-azure-function

For discoverability it's useful to have this in the Crate documentation. Still I wanted to leave this note here to that we can see if it's useful to combine them, merge them or something similiar.

File: docs/concepts/architecture-reference/azure-functions.rst [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+	  overwrites this paramter at run time, if it is set.
+	- ``connection`` signifies the name of the Azure Function application setting that contains our event hub's
+	  namespace connection string.
+	- ``cardinality`` signifies whether or not we want to use batching, or to consume messages one by one. Batching
+	  is far more efficient for our use case, which is why it is set to ``many`` instead of ``one``.
+	- ``consumerGroup`` signifies which consumer group will be used to consume messages from the eventhub. Ensure
+	  that it is unique, and not already being used by another consumer. If omitted, it will use the default consumer
+	  group of ``$Default``.
+
+For more information, please consule the `Azure Event Hubs bindings for Azure Functions documentation`_.
```
Comment (infoverload) [inline docs/concepts/architecture-reference/azure-functions.rst:179]: consule = consult

File: docs/concepts/architecture-reference/azure-functions.rst [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+	  the eventhub.
+	- ``eventHubName`` signifies the name of the event hub. The event hub name value in the connection string
+	  overwrites this paramter at run time, if it is set.
+	- ``connection`` signifies the name of the Azure Function application setting that contains our event hub's
+	  namespace connection string.
+	- ``cardinality`` signifies whether or not we want to use batching, or to consume messages one by one. Batching
+	  is far more efficient for our use case, which is why it is set to ``many`` instead of ``one``.
+	- ``consumerGroup`` signifies which consumer group will be used to consume messages from the eventhub. Ensure
+	  that it is unique, and not already being used by another consumer. If omitted, it will use the default consumer
+	  group of ``$Default``.
```
Comment (infoverload) [inline docs/concepts/architecture-reference/azure-functions.rst:177]: consider turning this into a table for easy parsing and removing redundancies such as 'signifies' 

recieving = receiving

run time = runtime

File: docs/concepts/architecture-reference/azure-functions.rst [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+
+For more information, please consule the `Azure Event Hubs bindings for Azure Functions documentation`_.
+
+local.settings.json
+...................
+
+To run this Azure Function application locallywe need to create a ``local.settings.json`` file
+at the root of our Azure Functions Application project. It should have the following contents:
```
Comment (infoverload) [inline docs/concepts/architecture-reference/azure-functions.rst:185]: locallywe = locally, you

our = your

contents = content

File: docs/concepts/architecture-reference/azure-functions.rst [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+The ``CrateConnectionString`` should be of the form postgres. MARK MARK
+
+``SinkTable`` is the name of the table we will be ingesting data into, with the ``SinkColumnUnmanned`` and
+``SinkColumnManned`` parameters defining what columns we will ingest unmanned and manned satellite data into,
+respectively.
+
+These settings **are not** deployed when we deploy the Azure Function. Instead, before we deploy, we should
+set up an Azure Function Application within the Azure Portal. You can do this `directly from VSCode`_, or
+using the `Azure CLI`_. These settings would then need be set up within the Azure Function's
+**Application Settings**.
```
Comment (infoverload) [inline docs/concepts/architecture-reference/azure-functions.rst:215]: we = you

need be = need to be

File: docs/concepts/architecture-reference/index.rst [hunk 5]
```diff
@@ -0,0 +1,12 @@
+=======================
+Reference Architectures
+=======================
+
+This section of the documentation covers ref archs.
```
Comment (infoverload) [inline docs/concepts/architecture-reference/index.rst:5]: ref archs = reference architectures

**Comment by autophagy** [comment]
@mechanomi @infoverload Thanks for the feedback! I will move this commit to the project we discussed with Johannes (How-To Guides) and then apply your changes.

**Pull request closed without merging**